packages/dma-contracts/contracts/actions/ajna/DepositBorrow.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.15;
+
+import { Executable } from "../common/Executable.sol";
+import { Write, UseStore } from "../common/UseStore.sol";
+import { OperationStorage } from "../../core/OperationStorage.sol";
+import { DepositBorrowData } from "../../core/types/Ajna.sol";
+import { AJNA_POOL_UTILS_INFO } from "../../core/constants/Ajna.sol";
+import { IAjnaPool } from "../../interfaces/ajna/IERC20Pool.sol";
+import { IAjnaPoolUtilsInfo } from "../../interfaces/ajna/IAjnaPoolUtilsInfo.sol";
+
+/**
+ * @title DepositBorrow | Ajna Action contract
+ * @notice Deposits collateral and borrows quoite token from Ajna pool
+ */
+contract AjnaDepositBorrow is Executable, UseStore {
+  using Write for OperationStorage;
+
+  constructor(address _registry) UseStore(_registry) {}
+
+  /**
+   * @param data Encoded calldata that conforms to the BorrowData struct
+   */
+  function execute(bytes calldata data, uint8[] memory) external payable override {
+    DepositBorrowData memory args = parseInputs(data);
```

Comment:
It might be needed to allow mapped values, and to sum deposit with mapped value as in aaveDeposit.